crates/subspace-networking/src/protocols/autonat_wrapper.rs

```diff
@@ -14,27 +14,39 @@ use tracing::debug;
 pub(crate) struct Config {
     pub(crate) inner_config: AutonatConfig,
     pub(crate) local_peer_id: PeerId,
+    pub(crate) servers: Vec<Multiaddr>,
 }
 
 pub(crate) struct Behaviour {
     inner: Autonat,
-    config: Config,
+    private_ips_enabled: bool,
     listen_addresses: HashSet<Multiaddr>,
 }
 
 impl Behaviour {
     pub(crate) fn new(config: Config) -> Self {
+        let mut inner = Autonat::new(config.local_peer_id, config.inner_config.clone());
+
+        for server in config.servers {
+            let maybe_peer_id = server.iter().find_map(|protocol| {
```

Comment:
This is a typical task and we have a function that can extract `PeerId` from a vec copy in utils: strip_peer_id. Please, either use this function or create a simplified version for future use.

Reply:
`strip_peer_id` primarily removes peer ID from multiaddress, we just need to extract it here. Do we really need a function for that?

Reply:
It's a very frequent task to extract `PeerId`, the function would be useful. Non-critical though.

Reply:
They have added nice `.with_p2p()` method in latest release, maybe they can add some more helper methods like this.